osv/semver_index.py

```diff
@@ -42,14 +42,26 @@ def _remove_leading_zero(component):
 def coerce(version):
   """Coerce a potentially invalid semver into valid semver."""
   version = _strip_leading_v(version)
-  version_pattern = re.compile(r'^(\d+)(\.\d+)?(\.\d+)?(.*)$')
+  version_pattern = re.compile(r'^(\d+)(\.\d+)?(\.\d+)?(([^+]*)(.*))$')
   match = version_pattern.match(version)
   if not match:
     return version
 
+  # also remove leading 0's from pre-release suffix
```

Comment:
Can you add a comment that this is to allow versions allow from earlier Semver 2.0.0 specs to be parsed as well?